docs/docs/examples/secure-api/pylon.mdx

```diff
@@ -0,0 +1,280 @@
+    const id = ctx.get("auth")!.sub;
+
+    return User.users.find((user) => user.id === id);
+  }
+
+  @requireAuth()
+  static async create() {
+    const ctx = getContext();
+
+    const auth = ctx.get("auth")!;
+
+    // Check if the user already exists
+
+    if (User.users.find((user) => user.id === auth.sub)) {
+      throw new ServiceError("User already exists", {
+        statusCode: 400,
+        code: "USER_ALREADY_EXISTS",
+      });
+    }
+
+    const user = new User(auth.sub, auth.username || "unknown", []);
+
+    User.users.push(user);
+
+    return user;
+  }
+}
+
+export default defineService({
+  Query: {
+    me: User.me,
+    info: () => "Public Data",
+  },
+  Mutation: {
+    createUser: User.create,
+  },
+});
+
+export const configureApp: PylonAPI["configureApp"] = (app) => {
+  // Initialize the authentication middleware
+  app.use("*", auth.initialize());
+
+  app.get("/api/info", (c) => {
+    return new Response("Public Data");
+  });
+
+  app.get("/api/me", auth.require(), (c) => {
+    const id = c.get("auth")!.sub;
+
+    const user = User.users.find((user) => user.id === id);
+
+    return c.json(user);
+  });
+
+  // A role check for `read:messages` is not required here, as the `user.messages` method already checks for it.
+  app.get("/api/me/mesages", auth.require({ roles: [] }), (c) => {
+    const id = c.get("auth")!.sub;
+
+    const user = User.users.find((user) => user.id === id);
+
+    if (!user) {
+      throw new HTTPException(404, { message: "User not found" });
+    }
+
+    // This will throw an error if the user does not have the `read:messages` role
+    return c.json(user.messages());
+  });
+};
+```
+
+### Call the API
+
+To call the API you need an access token, which is then verified by ZITADEL.
+Please follow [this guide here](/docs/guides/integrate/token-introspection/private-key-jwt#get-an-access-token), ignoring the first step as we already have the `.json`-key-file from the serviceaccount.
+
+:::info
+You can also create a PAT for the serviceuser and use it to test the API. For this, follow [this guide](/docs/guides/integrate/service-users/personal-access-token#create-a-service-user-with-a-pat).
+:::
+
+Optionally set the token as an environment variable:
+
+```
+export TOKEN='MtjHodGy4zxKylDOhg6kW90WeEQs2q...'
+```
+
+With the access token, you can then do the following calls:
+
+1. GraphQL:
+
+```
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={info}
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={me{id name}}
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/graphql?query={me{id name messages}}
+```
+
+You can also visit the GraphQL playground at `http://localhost:3000/graphql` and execute the queries there.
+
+2. Routes:
+
+```
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/api/info
+curl -H "Authorization: Bearer $TOKEN" -X GET http://localhost:3000/api/me
```

Comment:
Returns an empty response, no user data

Reply:
This is the case because the users are still empty. I will add a auto-creation of users after authenticating to make it more clear.